### PR TITLE
SRID support under MySQL 8.0 & compatible with 5.7

### LINKED ISSUE #9786: GeomFromText in MySQL changed to ST_GeomFromText
## What are you doing?
inserting into mysql db a geometry point

```js
const shulInfo = { ...req.body };
    shulInfo.location_point = {
        type: 'Point',
        coordinates: [shulInfo.location.lat, shulInfo.location.lng]
    };
    minyan_locations.create(shulInfo);
  })
```

## What do you expect to happen?
I wanted to insert

## What is actually happening?
Unhandled rejection SequelizeDatabaseError: FUNCTION godaven.GeomFromText does not exist
at Query.formatError (C:\projects\website\node_modules\sequelize\lib\dialects\mysql\query.js:247:16)
_Output, either JSON or SQL_

Dialect: mysql
__Dialect version:latest available
__Database version:latest available
__Sequelize version:latest available
__Tested with latest release:latest available

It worked when I changed your datatypes.js file to the following:
GEOMETRY.prototype._stringify = function _stringify(value, options) {
return 'ST_GeomFromText(' + options.escape(Wkt.convert(value)) + ')';
};

Previously it was "GeomFromText", I changed it to "ST_GeomFromText".
My reasoning was because MySQL now changed its syntax to "ST_GeomFromText".

### PR DESCRIPTION
**Pull request by kf6kjg**
### Pull Request check-list

- [yep] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [yep] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [yep] Have you added new tests to prevent regressions?
- [minor, but yep] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [can't] Did you update the typescript typings accordingly (if applicable)?
- [yep] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Closes #9786.

While it's true that MySQL 8.0 still isn't fully supported, several tests are failing under 8.0, this at least makes the SRID functionality work better.

The goal here was to be able to define and use columns that have SRIDs. All tests still pass under MySQL 5.7, and a few more pass under MySQL 8.0.

These changes, especially the ST_GeomFromText change, have been hacks the community has been doing for some time, as can be seen at https://github.com/sequelize/sequelize/issues/9786#issuecomment-474122602 - I've simply made them core and kept backwards compat.

### CODE REVIEW

**Comment by kf6kjg** [comment]
There are no existing tests I could find that test the GEOMETRY type's _stringify and _bindParam methods. Perhaps I missed them?  If not, where should they be added?

**Comment by sushantdhiman** [comment]
You can add tests [here](https://github.com/sequelize/sequelize/blob/8281de9082d0f65a6fe6a1d89157b259b97b4e95/test/integration/data-types.test.js#L385)

File: src/data-types.js [hunk 1]
```diff
@@ -830,10 +830,16 @@ class GEOMETRY extends ABSTRACT {
     this.srid = options.srid;
   }
   _stringify(value, options) {
-    return `GeomFromText(${options.escape(wkx.Geometry.parseGeoJSON(value).toWkt())})`;
+    if (this.srid) {
+      return `ST_GeomFromText(${options.escape(wkx.Geometry.parseGeoJSON(value).toWkt())}, ${this.srid})`;
```
Comment (sushantdhiman) [inline src/data-types.js:834]: From MySQL 5.7 https://dev.mysql.com/doc/refman/5.7/en/gis-wkt-functions.html

It seems `ST_GeomFromText` is synonyms for `GeomFromText`, I remember some time ago this was not true. Earlier in the main issue I asked for checking dialect version here, but that is no longer required.

Comment (kf6kjg) [reply src/data-types.js:834]: Yep, and as I also mentioned in the commit corresponding to this change:

> MariaDB added support for ST_GeomFromText in 5.3.3 according to https://mariadb.com/kb/en/mysqlmariadb-spatial-support-matrix/

**Comment by kf6kjg** [comment]
> You can add tests [here](https://github.com/sequelize/sequelize/blob/8281de9082d0f65a6fe6a1d89157b259b97b4e95/test/integration/data-types.test.js#L385)

It may take me a bit to get back to this: deep in crunch time until after mid-late this month trying to finish a project in time to get paid.

**Comment by blankstar85** [comment]
Any updates for this?

**Pull request closed without merging**